Wykonano zadanie

a-dreszler opened the pull request.

a-dreszler committed:
Wykonano zadanie

slawekludwiczak commented on src/main/java/model/Company.java:
```diff
@@ -0,0 +1,81 @@
+import static model.CompanyDepartaments.*;
+
+public class Company {
+    Employee[] employees;
```
private

slawekludwiczak commented on src/main/java/io/FileUtils.java:
```diff
@@ -0,0 +1,118 @@
+            if (employeesRead != lines) {
+                employees = Arrays.copyOf(employees, employeesRead);
+            }
+        } catch (FileNotFoundException | FileReadFailureException e) {
+            throw new FileReadFailureException("Nie znaleziono pliku: " + filePath, e);
+        }
```
Łapanie wyjątku FileReadFailureException w tym miejscu jest zbędne. Może on być rzucony z metody countLines(), gdzie wygląda dokładnie tak samo jak ten wyjątek, który wiersz niżej rzucasz z bloku catch. Zamiast go łapać i rzucać ponownie, niech ten oryginalny wyjątek leci sobie wyżej

slawekludwiczak commented on src/main/java/app/Main.java:
```diff
@@ -0,0 +1,20 @@
+            Company company = new Company(employees);
+            String stats = company.getCompanyStats();
+            FileUtils.saveToFile(stats, "stats.txt");
+        } catch (RuntimeException e) {
```
Staraj się nie chwytać tak ogólnych wyjątków. W tej sytuacji wiemy, jakie dokładnie wyjątki mogą być rzucone w bloku try, a jeżeli trafimy na coś, czego nie przewidzieliśmy, to należy to naprawić.
Łapanie RuntimeException, czy Exception niewiele mówi o tym co poszło nie tak

slawekludwiczak commented on src/main/java/io/FileUtils.java:
```diff
@@ -0,0 +1,118 @@
+            int employeesRead = 0;
+            employees = new Employee[lines];
+
+            for (int i = 0; input.hasNextLine(); i++) {
+                String[] splitEmployee = input.nextLine().split(";");
+                employees[i] = fillEmployee(splitEmployee);
+                if (employees[i] == null) {
+                    i--;
+                } else {
+                    employeesRead++;
+                }
```
Trochę bym tutaj odwrócił sytuację i zamiast najpierw zwiększać i, a później zmniejszać, to możesz i++ usunąć z pętli i przenieść do ifa:
```
for (int i = 0; input.hasNextLine(); ) {
...
    if (employees[i] != null) {
        i++;
        employeesRead++
    }
```

a-dreszler committed:
Wprowadzono poprawki:
 > Uproszczono wczytywanie pracowników metodą loadEmployeesFromFile
 > W metodzie main chwytane są już tylko wyjątki które faktycznie mogą wystąpić
 > Usunięto niepotrzebne łapanie wyjątku FileReadFailureException w metodzie loadEmployeesFromFile
